astropy/io/fits/header.py

```diff
@@ -105,7 +112,7 @@ def __init__(self, cards=[], txtfile=None):
             return
 
         if isinstance(cards, Header):
-            cards = cards.cards
+            cards = (copy.copy(card) for card in cards._cards)
```

Comment:
Just curious, is this guaranteed to always do a deep copy, or should you do ``copy.deepcopy(card)``?